src/orange/Types.hh

```diff
@@ -33,12 +33,12 @@ using VolumeId = OpaqueId<struct Volume>;
 /*!
  * Enumeration for cartesian axes.
  */
-enum class Axis
+enum Axis : size_type
 {
-    x,  //!< X axis/I index coordinate
-    y,  //!< Y axis/J index coordinate
-    z,  //!< Z axis/K index coordinate
-    size_  //!< Sentinel value for looping over axes
+    x = 0,  //!< X axis/I index coordinate
+    y = 1,  //!< Y axis/J index coordinate
+    z = 2,  //!< Z axis/K index coordinate
+    size_ = 3  //!< Sentinel value for looping over axes
```

Comment:
I'd like to avoid type-unsafe enums except in *very* special situations. One side effect of this is that `x` is now an enum value in the `celeritas` namespace, so that's very typo-prone.

Instead, what about helper functions
```c++
CELER_CONSTEXPR_FUNCTION int to_int(Axis a) { return static_cast<int>(a); }
inline CELER_FUNCTION Axis to_axis(int a) { CELER_EXPECT(a >= 0 && a < 3); return static_cast<Axis>(a); }
```